pkg/preflight/run_test.go

```diff
@@ -0,0 +1,219 @@
+//go:build integration
+
+// NOTE: requires a Kubernetes cluster in place currently, hence hidden behind a tag above
+// TODOLATER: get a mocked or ephemeral/Docker based K8s for use? see below some approaches which haven't played out yet
+// Test using: go test --tags=integration
+
+package preflight
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+	/*
+		See TODOLATER below
+
+		"k8s.io/client-go/kubernetes/fake"
+		k8sruntime "k8s.io/apimachinery/pkg/runtime"
+		discoveryfake "k8s.io/client-go/discovery/fake"
+	*/)
+
+func TestRunPreflights(t *testing.T) {
+	t.Parallel()
+
+	// A very simple preflight spec (local file)
+	_, testFilename, _, status := runtime.Caller(0)
+	if status == false {
+		t.Fatal("Cannot determine current directory for go test source code, required for relative path of preflight spec YAML")
+	}
+	preflightFile := fmt.Sprintf("%s/troubleshoot_v1beta2_preflight_gotest.yaml", filepath.ToSlash(filepath.Dir(testFilename)))
+
+	type preflightTests struct {
+		Interactive bool
+		Output      string
+		Format      string
+		Args        []string
+		//
+		ExpectedError      error
+		ExpectedOutputFile string
+		// May be in stdout or file, depending on above value
+		ExpectedOutputContent string
+	}
+
+	tests := []preflightTests{}
+	// TODO: test interactive true as well
+	for _, interactive := range []bool{false} {
```

Comment:
I did not review in detail this implementation in case you want to revise it to follow other table tests in the project.

Reply:
yea I think I will, thx. main observations so far:

- use of `github.com/stretchr/testify/assert` for less repetition, which makes sense eg. `assert.Equal()`
- the way the structs are defined is a bit less noisy in other tests as they are written now